packages/babel-plugin-transform-es2015-block-scoping/src/index.js

```diff
@@ -538,10 +538,20 @@ class BlockScoping {
     let fnPath;
     if (this.loop) {
       const ref = this.scope.generateUidIdentifier("loop");
-      const p = this.loopPath.insertBefore(
+      let transformedToBlockStatement = false;
+      if (this.loopPath.isStatementOrBlock()) {
+        transformedToBlockStatement = true;
+      }
+
+      let p = this.loopPath.insertBefore(
         t.variableDeclaration("var", [t.variableDeclarator(ref, fn)]),
       );
 
+      if (transformedToBlockStatement) {
```

Comment:
Why not just check if `p[0]` is a function?

Reply:
If #5710 gets merged I don't have to do anything here.